abdiff/config.py

```diff
@@ -1,4 +1,22 @@
 import logging
+import os
+from typing import Any
+
+
+class Config:
+    REQUIRED_ENV_VARS = ("WORKSPACE",)
+    OPTIONAL_ENV_VARS = ("DATA_DIRECTORY",)
+
+    def __getattr__(self, name: str) -> Any:  # noqa: ANN401
+        """Method to raise exception if required env vars not set."""
+        if name in self.REQUIRED_ENV_VARS or name in self.OPTIONAL_ENV_VARS:
+            return os.getenv(name)
+        message = f"'{name}' not a valid configuration variable"
+        raise AttributeError(message)
+
+    @property
+    def data_directory(self) -> str:
+        return self.DATA_DIRECTORY or "output"
```

Comment:
Could this just be `output_directory` for simplicity?

Reply:
Maybe pulling in [your comment below](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/2#discussion_r1731592951) @ehanson8: what if we named the property `root_working_directory`?  and updated the env var to `ROOT_WORKING_DIRECTORY`?  

Then we'd have the following throughout:
- `root_working_directory`: the base folder for Jobs to get created in
- `job_working_directory`: a Job specific folder, under the root working directory

Reply:
That works for me, just looking for symmetry and alignment between env var, args, and app variables whenever possible!

Reply:
Saw your comment after I wrote [my input](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/2#discussion_r1731650388), but I like what you've proposed above -- `root_working_directory` and `job_working_directory`.

Reply:
Thanks @ehanson8 and @jonavellecuerdo for the feedback. I agree it was a little awkward before.  Working on some changes now and will re-request a review.